packages/@lwc/engine-core/src/framework/secure-template.ts

```diff
@@ -29,17 +29,27 @@ export function registerTemplate(tpl: Template): Template {
 }
 
 /**
- * EXPERIMENTAL: This function acts like a hook for Lightning Locker
- * Service and other similar libraries to sanitize vulnerable attributes.
- * This API is subject to change or being removed.
+ * EXPERIMENTAL: This function acts like a hook for Lightning Locker Service and other similar
+ * libraries to sanitize vulnerable attributes.
  */
 export function sanitizeAttribute(
     tagName: string,
     namespaceUri: string,
     attrName: string,
     attrValue: any
-) {
-    // locker-service patches this function during runtime to sanitize vulnerable attributes.
-    // when ran off-core this function becomes a noop and returns the user authored value.
+): string {
+    // locker-service patches this function during runtime to sanitize vulnerable attributes. When
+    // ran off-core this function becomes a noop and returns the user authored value.
     return attrValue;
 }
+
+/**
+ * EXPERIMENTAL: This function acts like a hook for Lightning Locker Service and other similar
+ * libraries to sanitize HTML content. This hook process the content passed via the template to
+ * lwc:inner-html directive.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function sanitizeHtmlContent(content: unknown): string {
```

Comment:
Something worth noting that I didn't realize the first time when implementing this PR is that the `sanitizeAttribute` and `sanitizeHtmlContent` can't be overridden if the LWC engine is treated as a real ESM module.

It is impossible to reassign a named export from outside:

```js
import { sanitizeHtmlContent } from 'lwc';

sanitizeHtmlContent = content => content; 
// Uncaught TypeError: Assignment to constant variable.
```

```js
import * as lwc from 'lwc';

lwc.sanitizeHtmlContent = content => content; 
// Uncaught TypeError: Cannot assign to read only property 'sanitizeHtmlContent' of object '[object Module]'
```

We aren't facing this issue today because in places where we do override `sanitizeAttribute` LWC is consumed as a UMD module. We can either implement `sanitizeHtmlContent` correctly and fix `sanitizeAttribute` in the future or we can stick to the current pattern and fix both APIs later.

Thoughts?

Reply:
Correct @pmdartus, I also noticed it while modifying [this test](https://github.com/salesforce/lwc/pull/2256/files#diff-4c1b19c26186adbadb3762e71fd41a26cf8ab00fc7ee5a790ef1afcf14c606a0R61-R70); I was about to ask 😄. Let's implement `sanitizeHtmlContent` correctly!

When facing this issue, as a solution, I thought about exposing something like  `setSanitizeHtmlContentHook`. What do you have in mind?

Reply:
> When facing this issue, as a solution, I thought about exposing something like `setSanitizeHtmlContentHook`.

Perfect.